tfhe/src/integer/gpu/server_key/radix/vector_find.rs

```diff
@@ -0,0 +1,1672 @@
+use crate::core_crypto::gpu::lwe_ciphertext_list::CudaLweCiphertextList;
+use crate::core_crypto::gpu::CudaStreams;
+use crate::core_crypto::prelude::{LweBskGroupingFactor, UnsignedInteger};
+use crate::integer::block_decomposition::{BlockDecomposer, Decomposable, DecomposableInto};
+use crate::integer::gpu::ciphertext::boolean_value::CudaBooleanBlock;
+use crate::integer::gpu::ciphertext::{CudaIntegerRadixCiphertext, CudaUnsignedRadixCiphertext};
+use crate::integer::gpu::server_key::radix::CudaRadixCiphertext;
+use crate::integer::gpu::server_key::{CudaBootstrappingKey, CudaServerKey};
+use crate::integer::gpu::{
+    apply_univariate_lut_kb_async, compute_prefix_sum_hillis_steele_async, PBSType,
+};
+pub use crate::integer::server_key::radix_parallel::MatchValues;
+use crate::prelude::CastInto;
+use itertools::Itertools;
+use rayon::prelude::*;
+use std::hash::Hash;
+// /// MatchValues for the `match_value_parallelized` family of function
+
+impl CudaServerKey {
+    pub fn num_bits_to_represent_unsigned_value<Clear>(&self, clear: Clear) -> usize
```

Comment:
Should this be pub, or pub(crate) would be enough?

Reply:
i think they don't need to be public since we use them only here, maybe remove the pub is better